src/com/android/settings/bluetooth/BluetoothPairingController.java

```diff
@@ -226,6 +261,21 @@ boolean isLeContactSharingEnabled() {
         return mIsLeContactSharingEnabled;
     }
 
+    public boolean isHidDevice() {
+        return mIsHid;
+    }
+
+    public boolean getHidEnabledState() {
+        return mIsHidEnabled;
+    }
+
+    public void onDialogDismissed(boolean success) {
+        if (success) {
+            mBluetoothHidHost.setConnectionPolicy(mDevice,
+                    mIsHidEnabled ? CONNECTION_POLICY_ALLOWED : CONNECTION_POLICY_FORBIDDEN);
```

Comment:
```suggestion
                    mIsHid && mIsHidEnabled ? CONNECTION_POLICY_ALLOWED : CONNECTION_POLICY_FORBIDDEN);
```

This ensures that HID is always disabled if the device is not a HID device.